### PR TITLE
Add std::string::compare API to fixed_string [15351]

### PR DESCRIPTION
**Pull request by richiware**
## Description

Updating generated code from IDL files with last fastddsgen version breaks the compilation of several tests because
`fixed_string` doesn't implement `std::string::compare`. This PR implements this function.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [ ] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- [x] Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [ ] Documentation builds and tests pass locally. 
- [ ] New feature has been added to the `versions.md` file (if applicable).
- [ ] New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: include/fastrtps/utils/fixed_size_string.hpp [hunk 1]
```diff
@@ -170,6 +170,13 @@ struct fixed_string
         return string_len;
     }
 
+    /*!
+     * Compare with a std:.string.
```
Comment (jsantiago-eProsima) [inline include/fastrtps/utils/fixed_size_string.hpp:174]: Little typo: std::string instead of std:.string

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

File: include/fastrtps/utils/fixed_size_string.hpp (hunk 1 above)
Comment (richiware) [reply include/fastrtps/utils/fixed_size_string.hpp]: Done in a4d80e5

File: include/fastrtps/utils/fixed_size_string.hpp [hunk 2]
```diff
@@ -170,6 +170,19 @@ struct fixed_string
...
 
+    /*!
+     * Compare with a std::string.
+     *
+     * @param str std::string to be compared with.
+     *
+     * @return Integer value with the result of the comparison as described in `std.:string::compare()`.
+     */
+    int compare(
+            const std::string& str ) const
```
Comment (MiguelCompany) [inline include/fastrtps/utils/fixed_size_string.hpp:181]:
```suggestion
            const std::string& str) const
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
A test should be added to `test/unittest/utils/FixedSizeStringTests.cpp`

It would also be nice to have overloads for the compare method, similar to what we have for `operator ==`:

```
int compare(
        const char* s) const;

template<size_t N> int compare(
        const fixed_string<N>& str) const;
```

File: include/fastrtps/utils/fixed_size_string.hpp (hunk 2 above)
Comment (richiware) [reply include/fastrtps/utils/fixed_size_string.hpp]: Done in 6177ad3 and 83e0c96

**Comment by richiware** [comment]
@richiprosima Please test this

**Review by MiguelCompany — APPROVED** [review]
LGTM

**Comment by richiware** [comment]
Failed tests not related.

**Pull request merged**